actions/channel_actions.ts

```diff
@@ -104,15 +110,15 @@ export function searchMoreChannels(term, showArchivedChannels) {
             const myMembers = getMyChannelMemberships(state);
 
             // When searching public channels, only get channels user is not a member of
-            const channels = showArchivedChannels ? data : data.filter((c) => !myMembers[c.id]);
+            const channels = showArchivedChannels ? data : (data as Channel[]).filter((c) => !myMembers[c.id]);
```

Comment:
I wonder if there is any other way to type `data` to receive already the Channel[], instead of this casting. Maybe has dispatch some generic you can specify?

Reply:
Yes thats something we are hoping to improve in future, dispatch correctly return the data type. until then we are stuff with this